imjoy/services/ngrok.py

```diff
@@ -0,0 +1,95 @@
+import sys
+import os
+import platform
+import urllib.request
+import zipfile
+import stat
+import subprocess
+import json
+
+def detectHostArchitecture():
+        """Return the architecture as '386', 'amd64', 'arm32' or 'arm64'."""
+        out = ''
+        if platform.machine().lower()[:3] == 'arm':
+            out += 'arm'
+        if sys.maxsize > 2 ** 32:
+            if out == 'arm':
+                out += '64'
+            else:
+                out = 'amd64'
+        else:
+            if out == 'arm':
+                out += '32'
+            else:
+                out = '386'
+        return out
+
+def getCdnUrl():
+    env = os.environ.copy()
+    arch = env.get('NGROK_ARCH', (platform.system().lower() + '-' + detectHostArchitecture()))
+    cdn = env.get('NGROK_CDN_URL', 'https://bin.equinox.io')
+    cdnPath = env.get('NGROK_CDN_PATH', '/c/4VmDzA7iaHb/ngrok-stable-')
+    cdnFiles = {
+        'darwin-386':    cdn + cdnPath + 'darwin-386.zip',
+        'darwin-amd64':    cdn + cdnPath + 'darwin-amd64.zip',
+        'linux-arm32':        cdn + cdnPath + 'linux-arm.zip',
+        'linux-arm64':    cdn + cdnPath + 'linux-arm64.zip',
+        'linux-386':    cdn + cdnPath + 'linux-386.zip',
+        'linux-amd64':        cdn + cdnPath + 'linux-amd64.zip',
+        'win32-386':    cdn + cdnPath + 'windows-386.zip',
+        'win32-amd64':        cdn + cdnPath + 'windows-amd64.zip',
+        'freebsd-386':    cdn + cdnPath + 'freebsd-386.zip',
+        'freebsd-amd64':    cdn + cdnPath + 'freebsd-amd64.zip'
+    }
+
+    url = cdnFiles.get(arch)
+    if url is None:
+        raise Exception('ngrok - platform ' + arch + ' is not supported.')
```

Comment:
It's better to raise app specific exceptions. Ie create a base exception class that inherits from `Exception` that we can subclass into different exceptions as needed.

That will make it easier to differentiate exceptions.